lmfit/minimizer.py

```diff
@@ -282,11 +290,16 @@ def __residual(self, fvars):
         if self._abort:
             return None
         params = self.result.params
-        for name, val in zip(self.result.var_names, fvars):
-            params[name].value = params[name].from_internal(val)
+
+        if apply_scaling:
+            for name, val in zip(self.result.var_names, fvars):
+                params[name].value = params[name].from_internal(val)
+                params.update_constraints()
```

Comment:
Take `params.update_constraints` out of the `for` loop. It's an expensive call and it only needs to be called once in `__residual`.